plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStoragePoolManager.java

```diff
@@ -165,6 +166,20 @@ public boolean connectPhysicalDisksViaVmSpec(VirtualMachineTO vmSpec) {
     }
 
     public boolean disconnectPhysicalDisk(Map<String, String> volumeToDisconnect) {
+        if (MapUtils.isEmpty(volumeToDisconnect)) {
+            return false;
+        }
+
```

Comment:
The pool type is set in the StopCommand here, so it is better to use the respective adaptor to disconnect the volume (instead of looping through all the adaptors).

https://github.com/apache/cloudstack/blob/ac794a0c850480bd75837b003bd065b9a3d6e629/engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java#L1688-L1691